packages/aws-cdk-lib/aws-appsync/lib/appsync-function.ts

```diff
@@ -152,6 +167,10 @@ export class AppsyncFunction extends Resource implements IAppsyncFunction {
       throw new Error('Mapping templates cannot be used alongside code');
     }
 
+    if (props.maxBatchSize && !(props.dataSource instanceof LambdaDataSource)) {
+      throw new Error('Cannot set maxBatchSize when the data source is not LambdaDataSource');
```

Comment:
```suggestion
      throw new Error('maxBatchSize can only be set for the data source of type 'LambdaDataSource'');
```
This msg gives more info imo, let me know what you think

Reply:
Thanks. I fixed it.